src/Haskell/Ide/Engine/Plugin/Floskell.hs

```diff
@@ -31,20 +31,16 @@ floskellDescriptor plId = PluginDescriptor
 -- Formats the given source in either a given Range or the whole Document.
 -- If the provider fails an error is returned that can be displayed to the user.
 provider :: FormattingProvider
-provider uri typ _opts =
+provider contents uri typ _opts =
   pluginGetFile "Floskell: " uri $ \file -> do
     config <- liftIO $ findConfigOrDefault file
-    mContents <- readVFS uri
-    case mContents of
-      Nothing -> return $ IdeResultFail (IdeError InternalError "File was not open" Null)
-      Just contents ->
-        let (range, selectedContents) = case typ of
-              FormatDocument -> (fullRange contents, contents)
-              FormatRange r  -> (r, extractRange r contents)
-            result = reformat config (uriToFilePath uri) (BS.fromStrict (T.encodeUtf8 selectedContents))
-        in  case result of
-              Left  err -> return $ IdeResultFail (IdeError PluginError (T.pack err) Null)
-              Right new -> return $ IdeResultOk [TextEdit range (T.decodeUtf8 (BS.toStrict new))]
+    let (range, selectedContents) = case typ of
+          FormatDocument -> (fullRange contents, contents)
+          FormatRange r  -> (r, extractRange r contents)
+        result = reformat config (uriToFilePath uri) (BS.fromStrict (T.encodeUtf8 selectedContents))
```

Comment:
Why use `uriToFilePath` and not the `file` variable?

Reply:
No particular reason, it was like that when I found it.